modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/util/JournalUtil.java

```diff
@@ -413,6 +432,48 @@ private static Stack<JournalArticle> _getRecentArticles(
 		return recentArticles;
 	}
 
+	private static Layout _partialSearchLayout(
+		ThemeDisplay themeDisplay, PermissionChecker permissionChecker,
+		boolean privateLayout) {
+
+		boolean hasNextPart = true;
+
+		int start = 0;
+		int end = 0;
+		int interval = 20;
+
+		while (hasNextPart) {
+			end = start + interval;
+
+			List<Layout> layouts = LayoutLocalServiceUtil.getLayouts(
+				themeDisplay.getScopeGroupId(), privateLayout,
+				LayoutConstants.DEFAULT_PARENT_LAYOUT_ID, false, start, end);
```

Comment:
Is it necessary to set start and end?

Reply:
I thought that number of Layout of a site may be too large. We just need to find out the first Layout of all that satisfies the condition. So I think that the partial searching Layout will ignore loading too much data.